viadot/sources/s3.py

```diff
@@ -130,46 +140,88 @@ def from_df(
         Args:
             df (pd.DataFrame): Pandas DataFrame.
             path (str): Path to a S3 folder.
+            extension (Literal[".csv", ".parquet"]): Required file type. Accepted file formats are 'csv'
+                and 'parquet'. Defaults to '.parquet'.
+            max_rows_by_file (int): Max number of rows in each file (only works for read_parquet). Default to None.
         """
 
-        if path.endswith(".csv"):
+        if extension == ".csv":
             wr.s3.to_csv(
                 boto3_session=self.session,
                 df=df,
                 path=path,
-                dataset=True,
                 **kwargs,
             )
-        elif path.endswith(".parquet"):
+        else:
             wr.s3.to_parquet(
                 boto3_session=self.session,
                 df=df,
                 path=path,
-                dataset=True,
+                max_rows_by_file=max_rows_by_file,
                 **kwargs,
             )
-        else:
-            raise ValueError("Only CSV and parquet formats are supported.")
 
     def to_df(
         self,
-        path: str,
+        paths: list[str],
+        chunked: Union[int, bool] = False,
         **kwargs,
     ):
         """
         Reads a csv or parquet file to a pd.DataFrame.
 
         Args:
-            path (str): Path to a S3 folder.
+            paths (list[str]): A list of paths to S3 files.
+            chunked (Union[int, bool], optional): If True data will be split in a Iterable of DataFrames (Memory friendly).
+                If Integer data will be intereated by number of rows equal to the received Integer.
+
+        Example 1:
+        ```python
+        from viadot.sources import S3
+        s3 = S3()
+        s3.to_df(paths=['s3://{bucket}/pathfirstfile.parquet', 's3://{bucket}/pathsecondfile.parquet'])
+        ```
+
+        Example 2:
+        ```python
+        from viadot.sources import S3
+        s3 = S3()
+        dfs = s3.to_df(paths=['s3://{bucket}/pathfirstfile.parquet', 's3://{bucket}/pathsecondfile.parquet'], chunked = True/Integer)
+        for df in dfs:
+            print(df)
+        ```
         """
-        if path.endswith(".csv"):
+
+        if paths[0].endswith(".csv"):
```

Comment:
If you do that you should mention in the docstring that when user passes a list of files, they must all be in the same format. Otherwise this could lead to user getting random error.

Reply:
Updated in:

- [📝 Fixed Docstring for file format](https://github.com/dyvenia/viadot/pull/603/commits/cec960e6d5e9f4cc425959c5964c5fb718973b5f)